shm/src/map.cpp

```diff
@@ -17,6 +17,7 @@ Map::Map() {
         }
         Islands_.push_back(temp);
     }
+    currentPosition_ = &(*std::next(Islands_.begin()));
```

Comment:
Probably we could, but as you mentioned, it will prone to some type conflicts - I'm not saying that you are not right, but probably we will make some refactor, when we have all tasks done.